Product Collection: move "Display Settings" from Toolbar to Inspector Controls

imanish003 opened the pull request:
<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #52622

Similar to PR https://github.com/WordPress/gutenberg/pull/58207 in Gutenberg repo, this PR moves the "Display Layout" menu from the block Toolbar to the Inspector Controls panel. This change improves the discoverability of display settings and aligns with the pattern used in core blocks.

The changes include:
- Moving the display layout controls from the block toolbar to the Inspector Controls panel
	- All of display settings are moved to "Settings" section
	- Keeping "Offset" and "Max pages to show" hidden under three dots menu
	- Placing "Products per page" under "Columns" control as they're related
- Removing the toolbar "Display Settings"
- Removing "Dimensions" section for now as discussed in Slack (https://a8c.slack.com/archives/C05UMTA7W0P)

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

**Case 1: Verify UI changes in Inspector Controls**
1. Create a new post and add a Product Collection block.
2. Click on "create your own" to add Custom Collection. 
3. Verify that "Products per page" is available in "Settings" section.
4. Verify that by default both "Offset" and "Max pages to show" are hidden i.e. not visible in Settings section.
5. Click on three dots menu and verify that "Offset" and "Max pages to show" available in dropdown menu.
6. Verify that you are able to add "Offset" and "Max pages to show" to Settings section by clicking on three dots menu and selecting these options.
<img width="280" alt="image" src="https://github.com/user-attachments/assets/10cb0770-afc5-44e4-9518-22f31f1d1927">

6. Verify that there is no "Dimensions" section in Inspector controls. 
	- "Width" control (Fill and Fixed) should be available in "Settings" section.

**Case 2: "Products Per Page" control"**
1. Change "Products Per Page" value, for example 2.
2. As soon as you change the value, it should update number of products on Editor side. In our case, 2 products should be shown.
3. Save the changes.
4. Verify on Frontend that number of products is 2.

**Case 3: "Offset" control**
1. Add "Offset" control by clicking on three dots and selecting "Offset".
2. Test offset functionality:
   - Set the "Offset" value to 2
   - Verify in the editor that:
     - The first 2 products are skipped
     - The product list starts from the 3rd product
   - The changes should be reflected immediately without needing to save

3. Verify frontend behavior:
   - Save the post
   - View the post on the frontend
   - Confirm that the offset is working correctly:
     - First 2 products are skipped
     - Product list starts from the 3rd product
     - The display matches what was shown in the editor

**Case 4: "Max Pages to Show" control**
1. Add "Max Pages to Show" control by clicking on three dots and selecting "Max Pages to Show"
2. Set "Products Per Page" to 2 to ensure multiple pages
3. Set "Max Pages to Show" to 2
3. Verify frontend behavior:
   - Save the post
   - View the post on the frontend
   - Confirm that:
     - Only 2 page numbers are displayed in pagination
     - You can still navigate between available pages


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
Product Collection: move "Display Settings" from Toolbar to Inspector Controls
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

imanish003 committed:
Move display layout menu from toolbar to inspector controls

This commit moves the display layout menu from the toolbar to the inspector
controls panel. The display settings include:

- Products per page
- Offset
- Max pages to show

The display settings are only shown when the query is not inherited from the
template.

imanish003 committed:
Delete display settings toolbar

The display settings toolbar has been removed as these controls are being moved
to the inspector panel for better UX.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks

imanish003 committed:
Merge branch 'trunk' of https://github.com/woocommerce/woocommerce into update/52622-product-collection-move-display-layout-menu-from-toolbar-to-inspector-controls

imanish003 committed:
E2E Tests: Move display layout menu from toolbar to inspector controls

The display layout menu has been moved from the toolbar to the inspector controls.
This commit updates the tests to reflect this change.

imanish003 committed:
Merge branch 'trunk' into update/52622-product-collection-move-display-layout-menu-from-toolbar-to-inspector-controls

imanish003 committed:
Fix failing test

kmanijak commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/display-settings.tsx:
Considering that those files:

```
import ProductsPerPageControl from './products-per-page-control';
import OffsetControl from './offset-control';
import MaxPagesToShowControl from './max-pages-to-show-control';
```

includes only a "supportive" blocks and are not used in the `inspector-controls/index.tsx` I'd suggest extracting this file (`plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/display-settings.tsx`) and those dependency files to dedicated directory `inspector-controls/display-settings/*` and then renaming this file to `index.tsx`.

Reasoning:
1. That would be in sync with ``inspector-controls/price-range` and `inspector-controls/taxonomy-controls` that have more than 1 file.
2. It would increase the readability of `inspector-controls` directory which is already pretty dense and full of files.

Not a must but a suggestion!

imanish003 replied:
I believe this isn't needed anymore, as we have removed the `display-settings.tsx` file, which isn't needed anymore after we moved display settings to Inspector Controls.

kmanijak reviewed (changes requested):
Thanks for working on this! It's testing well, controls works as expected.

I know it wasn't agreed upfront, so sorry about that, but one thing I'd like us to change is to move display controls to "Settings" section and:
- keep "Offset" and "Max pages to show" hidden
- place "Products per page" under "Columns" control as they're related.

(Also, raised elsewhere: p1731670836482299-slack-C05UMTA7W0P)

imanish003 committed:
Move display settings to Settings section in inspector controls

- Move display controls to "Settings" section
- Keep "Offset" and "Max pages to show" hidden
- Place "Products per page" under "Columns" control as they're related.

imanish003 committed:
Extract query reset logic from getDefaultSettings into a separate function

imanish003 committed:
Remove Dimensions section and move width control to Settings section

imanish003 commented:
Hey @kmanijak! That's good feedback. I agree that it would be better to move display settings to the "Settings" section.

I have made the changes as suggested i.e.
- Moved display controls to "Settings" section. Which means I removed "Display" section entirely.
- Kept "Offset" and "Max pages to show" hidden
- Placed "Products per page" under "Columns" control as they're related.
- Removed "Dimensions" section for now as discussed in Slack.

Can you please review the PR again?

kmanijak commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx:
```diff
@@ -154,22 +165,20 @@ const ProductCollectionInspectorControls = (
 				) }
 				<LayoutOptionsControl { ...displayControlProps } />
+				<WidthOptionsControl { ...dimensionsControlProps } />
 				<ColumnsControl { ...displayControlProps } />
+				{ showProductsPerPageControl && (
+					<ProductsPerPageControl { ...queryControlProps } />
+				) }
 				{ showOrderControl && (
 					<OrderByControl { ...queryControlProps } />
 				) }
```
Sorry for being picky here but I'd suggest changing the controls order so the range controls are next to each other. I think it's easier to digest the information if the same type of controls are next to each other

```suggestion
				{ showProductsPerPageControl && (
					<ProductsPerPageControl { ...queryControlProps } />
				) }
				<ColumnsControl { ...displayControlProps } />
				{ showOrderControl && (
					<OrderByControl { ...queryControlProps } />
				) }
```

Current | My suggestion
<img width="283" alt="image" src="https://github.com/user-attachments/assets/a7c73132-7f6d-4f68-a5bd-6f608feb4136"> | <img width="271" alt="image" src="https://github.com/user-attachments/assets/fb7c2d67-7d0d-480e-a129-0ccaad80afeb">

imanish003 replied:
Thanks Karol! I appreciate your attention to detail. I have made the changes in 804041c

kmanijak reviewed (changes requested):
I left one more ask for rearranging the controls. Feel free to ignore it if you don't agree 🙌 

One last feedback is the difference in styling between Products per Page and Columns (input box has different size). Can we make them look the same?

<img width="275" alt="image" src="https://github.com/user-attachments/assets/c87e254a-c85f-4576-9d9a-f060a58b3725">

imanish003 committed:
Make Products per Page and Columns input boxes consistent in size

The Products per Page and Columns input boxes had different sizes in the inspector controls. This commit makes them consistent by adding the __next40pxDefaultSize prop to the Columns RangeControl component to match the Products per Page control.

imanish003 committed:
Move range controls next to each other in inspector controls

Reorders the inspector controls to group similar control types together,
specifically placing the range controls (Products per page and Columns)
adjacent to each other for better UX and information organization.

imanish003 commented:
> One last feedback is the difference in styling between Products per Page and Columns (input box has different size). Can we make them look the same?

That's a good catch. I have made the Products per Page and Columns input boxes consistent in size.

Thanks for the feedback. Can you please review the PR again?

imanish003 committed:
Merge branch 'trunk' into update/52622-product-collection-move-display-layout-menu-from-toolbar-to-inspector-controls

imanish003 committed:
Fix E2E test

kmanijak reviewed (approved):
Thanks for your patience in addressing all the comments! Looking good! 🙌
